torch/utils/data/_utils/worker.py

```diff
@@ -223,6 +223,14 @@ def _worker_loop(dataset_kind, dataset, index_queue, data_queue, done_event,
         seed = base_seed + worker_id
         random.seed(seed)
         torch.manual_seed(seed)
+        for generator in torch.random._generator_registry:
```

Comment:
It occurs to me, technically, you don't even need the generator registry; we could just gc.get_objects() and traverse the entire live heap to look for generators 😆

Reply:
Actually, legit question, why don't we do this? The patch becomes super simple then.

Reply:
I gave it a try in https://github.com/pytorch/pytorch/pull/107131, LMK what you think

Reply:
Ho yeah that definitely makes it a a lot less invasive if we're ok with traversing all the alive objects on process creation. Which I think we are.